app/test/consistent_apphash_test.go

```diff
@@ -3,134 +3,300 @@ package app_test
 	}
+	feegrantMsg, err := feegrant.NewMsgGrantAllowance(&basicAllowance, accountAddresses[0], accountAddresses[1])
+	require.NoError(t, err)
+	firstBlockSdkMsgs = append(firstBlockSdkMsgs, feegrantMsg)
 
-	// Create a Blob Tx
-	blobTx := BlobTx{
-		author:    accountNames[2],
-		blobs:     []*blob.Blob{blob.New(fixedNamespace(), []byte{1}, appconsts.DefaultShareVersion)},
-		txOptions: blobfactory.DefaultTxOpts(),
+	// NewMsgSubmitProposal - submits a proposal to send funds from the governance account to account-1
+	govAccount := testApp.GovKeeper.GetGovernanceAccount(testApp.NewContext(false, tmproto.Header{})).GetAddress()
+	msgSend := banktypes.MsgSend{
+		FromAddress: govAccount.String(),
+		ToAddress:   accountAddresses[1].String(),
+		Amount:      amount,
 	}
+	proposal, err := govtypes.NewMsgSubmitProposal([]sdk.Msg{&msgSend}, amount, accountAddresses[0].String(), "")
+	require.NoError(t, err)
+	firstBlockSdkMsgs = append(firstBlockSdkMsgs, proposal)
 
-	// Create SDK Tx
-	rawSdkTx, err := signer.CreateTx(sdkTx.sdkMsgs, sdkTx.txOptions...)
+	// NewMsgDeposit - deposits funds to a governance proposal
+	msgDeposit := govtypes.NewMsgDeposit(accountAddresses[0], 1, depositAmount)
+	firstBlockSdkMsgs = append(firstBlockSdkMsgs, msgDeposit)
+
+	// NewMsgCreateValidator - creates a new validator
+	msgCreateValidator, err := stakingtypes.NewMsgCreateValidator(sdk.ValAddress(accountAddresses[6]),
+		ed25519.GenPrivKeyFromSecret([]byte("validator")).PubKey(),
+		amount[0],
+		stakingtypes.NewDescription("taco tuesday", "my keybase", "www.celestia.org", "ping @celestiaorg on twitter", "fake validator"),
+		stakingtypes.NewCommissionRates(sdk.NewDecWithPrec(6, 0o2), sdk.NewDecWithPrec(12, 0o2), sdk.NewDecWithPrec(1, 0o2)),
+		sdk.OneInt())
 	require.NoError(t, err)
+	firstBlockSdkMsgs = append(firstBlockSdkMsgs, msgCreateValidator)
+
+	// NewMsgDelegate - delegates funds to validator-0
+	msgDelegate := stakingtypes.NewMsgDelegate(accountAddresses[0], genValidators[0].GetOperator(), amount[0])
+	firstBlockSdkMsgs = append(firstBlockSdkMsgs, msgDelegate)
+
+	// NewMsgBeginRedelegate - re-delegates funds from validator-0 to validator-1
+	msgBeginRedelegate := stakingtypes.NewMsgBeginRedelegate(accountAddresses[0], genValidators[0].GetOperator(), genValidators[1].GetOperator(), amount[0])
+	firstBlockSdkMsgs = append(firstBlockSdkMsgs, msgBeginRedelegate)
+
+	// ------------ Second Block ------------
+
+	var secondBlockSdkMsgs []sdk.Msg
+
+	// NewMsgVote - votes yes on a governance proposal
+	msgVote := govtypes.NewMsgVote(accountAddresses[0], 1, govtypes.VoteOption_VOTE_OPTION_YES, "")
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgVote)
+
+	// NewMsgRevoke - revokes authorization from account-1
+	msgRevoke := authz.NewMsgRevoke(
+		accountAddresses[0],
+		accountAddresses[1],
+		blobtypes.URLMsgPayForBlobs,
+	)
+
+	// NewMsgExec - executes the revoke authorization message
+	msgExec := authz.NewMsgExec(accountAddresses[0], []sdk.Msg{&msgRevoke})
```

Comment:
This is framed weird. 

User A should grant permission to User B
User B should use MsgExec to perform the action
User A should revoke User B's permission

But here I believe you have User B performing an action of User A's behalf to revoke permission to execute messages to themself. Maybe it's not that problematic, it's like User B is saying no thank you, I don't want your authority